### DISCUSSION TITLE
how to implement bg color animation  like githubnext page

### DISCUSSION THREAD

**Discussion by xwyzsn** [tailwindlabs/tailwindcss#9846] [Help] [answered]
i notice that githubnext is developed using tailwind,but i just wonder know how can i achieve this animation. 
when i hover a card the background color changed.
![image](https://user-images.githubusercontent.com/55912637/201957502-91186993-019e-48d3-8149-e9484f0cefd2.png)

**Comment by wongjn** [comment, marked as answer]
Here is the HTML within the cards that is responsible for this effect:
```html
<div class="absolute top-0 bottom-0 right-0 w-[max(30em,50%)] opacity-80">
  <svg class="absolute inset-0 w-full h-full" viewBox="0 0 1 1" preserveAspectRatio="none">
    <defs>
      <linearGradient id="hey-github" x1="0" y1="1" x2="1" y2="0">
        <stop offset="0" stop-color="#8AE8FF"></stop>
        <stop offset="1" stop-color="#6E56CF"></stop>
      </linearGradient>
    </defs>
    <path d="M0 1 Q 0.8 0.8 1 0 L 1 1 Z" fill="url(#hey-github)"></path>
  </svg>
  <div class="absolute inset-0 bg-white transition duration-300 group-hover:translate-x-[120%] will-change-transform"></div>
</div>
```
They have an `<svg>` that creates the "swoop" shape and then in the default state, they have a white rectangle `<div>` element stacked on top of the `<svg>` element in the z direction such that the `<svg>` element is behind the `<div>` element. When one hovers over the `group` card HTML element (not included in the above snippet), the white element moves to to the right, revealing the shape.

**Reply by xwyzsn** [reply]
thank you!!! learned a lot
